activerecord/test/cases/connection_adapters/connection_handlers_sharding_db_test.rb

```diff
@@ -316,6 +316,9 @@ def test_default_shard_is_chosen_by_first_key_or_default
 
         assert_equal :not_default, SecondaryBase.default_shard
         assert_equal :default, SomeOtherBase.default_shard
+      ensure
+        name = "ActiveRecord::ConnectionAdapters::ConnectionHandlersShardingDbTest::SecondaryBase"
+        ActiveRecord::Base.connection_handler.remove_connection_pool(name, role: :writing, shard: :not_default)
```

Comment:
We have to remove this one because it uses a different shard name than the other tests which would cause tests checking the connection names on the handler to be flaky because this connection was still left over.